Add notes for 2022-{04-04, 05-16, 06-06, 08-15}, touch up existing notes (#291)
unicode-org/message-format-wg: meetings/2022/notes-2022-08-15.md

- Romulo Cintra (RCA) Igalia
- David Filip - XLIFF TC, Huawei (DAF)
- Eemeli Aro - Mozilla (EAO)
- Mihai Nita - Google (MIH)
- Elango Cheran - Google (ECH)
- Zibi Braniecki - Amazon (ZBI)
- Richard Gibson - OpenJSF (RGN)

## MessageFormat Working Group Contacts: 

- [Mailing list](https://groups.google.com/a/chromium.org/forum/#!forum/message-format-wg)

### Agenda
 
- Past meeting Actions 
- Implementation Status
- Review PR’s & Open Issues
- Message resource WG 
 
 
 
## Past meeting Actions 
- [  ] Consensus about what we should do next regarding meeting cadence,  should we  cancel extended meetings ?
 
RCA: Can we revisit whether we need our weekly extended meetings?
 
EAO: Let's revisit this question later in the meeting.
 
 
## Implementation Status
 
MIH : I have a design doc and it’s posted for review and it’s how public API should look for ICU, about implementation. I have something working, and I’m using a JSON file provided by EAO to analyze how the JS implementation is working on tests.
 
I found some parts that I don’t think we should support due to the fact that they aren't specified.
 
As soon as I have a more “stable” implementation I will share the PR.
 

> Link to the JSON test data: https://github.com/messageformat/messageformat/blob/master/packages/mf2-messageformat/src/__fixtures/test-messages.json
>  
> EAO: The [JSON](https://github.com/messageformat/messageformat/blob/master/packages/mf2-messageformat/src/__fixtures/test-messages.json) test data is entirely authored by me. It's an Apache2 license. If there is a place that I can publish it that is more usable, then I am happy to do it.  

Reviewer: ```suggestion
EAO: The [JSON test data](https://github.com/messageformat/messageformat/blob/master/packages/mf2-messageformat/src/__fixtures/test-messages.json) is entirely authored by me. It currently has an Apache2 license. If there is a place that I can publish it that is more usable, then I am happy to do it.  
```

 
ECH : Can you put it in the repo?
 
EAO : I can add it and modify license accordingly with needs, I can PR against `main` or `develop`, what do you think? 
 
MIH: I would say put it in the `develop` branch along with the working spec. Once the spec is ready, we merge it to `main`.
 
EAO: Some parts are specific to my implementation, like testing for error codes, which I don't think is relevant to the common spec.
 
MIH: I did some minor modifications. If in your test is an error array and gets an error it just passes, I consider no error.
 
EAO: That is correct.
 
RCA: What is the status of the implementation work?
 

> EAO: The JS work is effectively done.  I have tagged the NPM package as ___.

Reviewer: ```suggestion
EAO: The JS work is effectively done, and it's available on npm as [messageformat@next](https://www.npmjs.com/package/messageformat/v/next).
```

> There are some details about inconsistencies found but aren’t published yet (how markup it’s handled and how it’s stringified to represent black box we agreed). There is some tooling that I haven't yet released, like the 2-way sync between Fluent, and also the ability to parse ICU MessageFormat 1 and Fluent messages by the MessageFormat v2 runtime. Please give it a try.  It implements the API in the proposal for JS Intl.MessageFormat that I am proposing.

Reviewer: ```suggestion
There are some details about inconsistencies found but aren’t published yet (how markup it’s handled and how its stringification is black-boxed as we agreed). Tooling is also available to parse ICU MessageFormat 1 and Fluent messages for use by the MessageFormat 2 runtime. There are some updates that I haven't yet released, like the 2-way sync between MF2 and Fluent. Please give it a try.  It implements the API in the ECMA-402 `Intl.MessageFormat` proposal for JS that I am championing.
```

 
RCA: I think this deserves a blog post to announce the work that you have done.
 
ECH: I think it’s really interesting to get more feedback and see how people in general feel.
 
MIH: 
 
```
let $foo = {$count :number currency=$cur precision=2 rounding=up}
let $pers = {$foo :person level=formal}
let $bar = {$pers :date skeleton=yMMMdE}
match {$bar}
   when 1 {...}
   when one {...}
   when male {...}
   when foggy {...}
   when * {...}
```
 
Here is an example of our current working syntax.  Something like this is legal, and I saw an instance of this in our JSON test data. However, I think this is confusing for a user, and it is not clear what the expected behavior should be.
 
EAO: It might be more helpful to paste the data from JSON to be more direct trying to solve the problem
 
MIH: This is not a strawman argument, I have to implement.
 

> EAO: I don't think anyone is asking you to implement a cast to a date.

Reviewer: ```suggestion
EAO: I don't think anyone is asking you to implement a cast from a person to a date.
```

 
MIH: Is that a cast?
 
EAO: There should be an example where there are 2 local variables, and one depends on the other, and one is being used as a selector.
 
MIH: 
```

> let $foo = {$bar} match {$foo} when one {one} when * {other}

Reviewer: ```suggestion
let $foo = {$bar}
match {$foo}
when one {one}
when * {other}
```
